packages/jupyter-chat/src/types.ts

```diff
@@ -87,21 +87,70 @@ export interface INewMessage {
 }
 
 /**
- * The attachment interface.
+ * The attachment type. Jupyter Chat allows for two types of attachments
+ * currently:
+ *
+ * 1. File attachments (`IFileAttachment`)
+ * 2. Cell attachments (`ICellAttachment`)
+ *
+ * The `type` field is always defined on every attachment, so it can be used to
+ * distinguish different attachment types.
  */
-export interface IAttachment {
+export type IAttachment = IFileAttachment | ICellAttachment;
```

Comment:
I think I would make the two top level ones `IFileAttachment | INotebookAttachment` and then have the cell part an optional child of `INotebookAttachment`. That way, the top level ones are always file types.